src/stdlib/builtin/lv_string_builtin.c

```diff
@@ -232,6 +232,18 @@ int lv_strcmp(const char * s1, const char * s2)
     return *(const unsigned char *)s1 - *(const unsigned char *)s2;
 }
 
+int lv_strncmp(const char * s1, const char * s2, size_t len)
+{
+    while(len > 0 && *s1 && (*s1 == *s2)) {
```

Comment:
This function assumes that we want to compare s1 to s2 and not the other way. Please indicate this or handle corner cases.

Reply:
There are 3 different cases:
1. The length of s1 is equal to the length of s2
2. The length of s1 is less than the length of s2
3. The length of s1 is greater than the length of s2

In the first case the while loop will break at then end of both strings because `*s1` will be false when we reach `\0`
In the second case the comparison will stop again when we reach the null terminator.
In the third case the comparison will stop because the last character of s2 (`\0`) is different from the nonzero character in s1

The comparison will stop anyways if `len` is shorter than the length of the shortest string.

Am I misunderstanding something?